src/packages/@ncigdc/modern_components/ClinicalAggregations/ClinicalAggregations.tsx

```diff
@@ -0,0 +1,535 @@
+}
+
+interface INotificationProps {
+  components: string[],
+  level: string,
+  id: string,
+  dismissible: boolean,
+  message: JSX.Element,
+  dismissed?: boolean,
+}
+
+interface IClinicalProps {
+  filteredFacets: IfilterdFacetsProps,
+  theme: ITheme,
+  setUselessFacetVisibility: (uselessFacetVisibility: boolean) => void,
+  shouldHideUselessFacets: boolean,
+  searchValue: string,
+  setSearchValue: (searchValue: string) => void,
+  handleQueryInputChange: () => void,
+  parsedFacets: IParsedFacetsProps | {},
+  isLoadingParsedFacets: boolean,
+  allExpanded: { [x: string]: boolean },
+  facetsExpandedStatus: IExpandedStatusStateProps,
+  dispatch: (action: IExpandedStatusActionProps) => void,
+  notifications: INotificationProps[],
+  maxFacetsPanelHeight: number,
+}
+
+interface IFieldProps {
+  __dataID: string,
+  customSort: number,
+  description: string,
+  name: string,
+  type: { name: string, __dataID: string },
+}
+
+const headersHeight = 44;
+
+const facetMatchesQuery = (
+  facet: IFacetProps,
+  elements: IBucketProps[],
+  searchValue: string,
+): boolean => {
+  return some(
+    [
+      replace(facet.field.split('.').pop() || '', /_/g, ' '),
+      ...(elements || []).map((e: IBucketProps) => e.key),
+      facet.description,
+    ]
+      .filter((n: string) => n)
+      .map(toLower),
+    searchTarget => includes(
+      searchTarget.toLocaleLowerCase(),
+      searchValue.toLocaleLowerCase(),
+    ),
+  );
+};
+
+const MagnifyingGlass = styled(SearchIcon, {
+  backgroundColor: ({ theme }: { theme: ITheme }) => theme.greyScale5,
+  border: ({ theme }: { theme: ITheme }) => `1px solid ${theme.greyScale4}`,
+  borderRadius: '4px 0 0 4px',
+  borderRight: 'none',
+  color: ({ theme }: { theme: ITheme }) => theme.greyScale2,
+  height: '3.4rem',
+  padding: '0.8rem',
+  width: '3.4rem',
+});
+
+const enhance = compose<IClinicalProps, IClinicalProps>(
+  setDisplayName('ClinicalAggregations'),
+  withTheme,
+  connect(
+    ({
+      bannerNotification,
+      facetsExpandedStatus,
+    }: {
+      facetsExpandedStatus: IExpandedStatusStateProps,
+      bannerNotification: INotificationProps[],
+    }) => ({
+      allExpanded: mapValues(facetsExpandedStatus, status => some(values(status.facets))),
+      facetsExpandedStatus,
+      notifications: bannerNotification,
+    }),
+  ),
+  withState('isLoadingParsedFacets', 'setIsLoadingParsedFacets', false),
+  withState('shouldHideUselessFacets', 'setShouldHideUselessFacets', true),
+  withState('searchValue', 'setSearchValue', ''),
+  withFacetSelection({
+    entityType: 'ExploreCases',
+    presetFacetFields: presetFacets.map(x => x.field),
+    validFacetDocTypes: ['cases'],
+    validFacetPrefixes: [
+      'cases.demographic',
+      'cases.diagnoses',
+      'cases.diagnoses.treatments',
+      'cases.exposures',
+      'cases.follow_up',
+      'cases.molecular_tests',
+    ],
+  }),
+  withPropsOnChange(
+    ['customCaseFacets', 'globalFilters'],
```

Comment:
Since these props are objects, I believe `withPropsOnChange` doesn't check appropriately for these if just listed as an array of strings, therefor the use of lodash's `isEqual` in other components.

Reply:
Good question. I'll double check